components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2262,17 +2213,29 @@ public Group updateGroup(Group oldGroup, Group newGroup, Map<String, Boolean> re
                         if (oldMembers != null && oldMembers.contains(newMember)) {
                             continue;
                         }
+                        doUserValidation(newMember, userStoreDomainForGroup, oldGroup.getDisplayName(),
+                                newlyAddedUserIds);
                         addedMembers.add(newMember);
                     }
                 }
+            }
 
-                if (CollectionUtils.isNotEmpty(addedMembers) || CollectionUtils.isNotEmpty(deletedMembers)) {
-                    carbonUM.updateUserListOfRole(newGroup.getDisplayName(),
-                            deletedMembers.toArray(new String[deletedMembers.size()]),
-                            addedMembers.toArray(new String[addedMembers.size()]));
-                    updated = true;
-                }
+            // We do not update Identity_SCIM DB here since it is updated in SCIMUserOperationListener's methods.
+            // Update name if it is changed
+            if (!(oldGroup.getDisplayName().equalsIgnoreCase(newGroup.getDisplayName()))) {
```

Comment:
You could assign oldGroup.getDisplayName() & newGroup.getDisplayName() and reuse them since it is being used more than one place

Reply:
Fixed